# Time of Submission is identical for many of these FCC public comment filings. These are possiby fake

In [76]:
# Multiple sumbissions at the exact same instant.. possibly automatically generated
print(df_500k_zip.date_submission.value_counts())

2017-05-24T23:18:04+00:00    86237
2017-05-24T23:09:20+00:00    81837
2017-05-24T23:14:50+00:00    61642
2017-05-24T23:11:57+00:00    38261
2017-05-24T23:19:47+00:00    30394
2017-05-25T10:01:58+00:00    13403
2017-05-24T22:58:31+00:00    12958
2017-05-25T10:01:36+00:00    12858
2017-05-24T23:21:10+00:00    12319
2017-05-25T10:01:15+00:00    12245
2017-05-25T10:03:36+00:00     9432
2017-05-25T10:02:58+00:00     9029
2017-05-25T10:00:52+00:00     8257
2017-05-25T10:02:15+00:00     7767
2017-05-25T10:04:06+00:00     7473
2017-05-25T10:02:36+00:00     7151
2017-05-25T10:00:32+00:00     6213
2017-05-25T09:59:45+00:00     5249
2017-05-25T10:00:10+00:00     5246
2017-05-25T10:03:50+00:00     4099
2017-05-25T10:03:18+00:00     3776
2017-05-25T21:15:41.088Z         3
2017-05-25T20:06:31.317Z         3
2017-05-25T20:36:58.727Z         3
2017-05-25T20:30:19.853Z         3
2017-05-25T14:23:11.888Z         2
2017-05-25T20:42:26.816Z         2
2017-05-25T20:36:04.326Z         2
2017-05-25T14:40:51.

In [1]:
# imports:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from wordcloud.wordcloud import WordCloud

In [2]:
df_500k_zip = pd.read_csv('FCC_comments_500K.csv', encoding='utf8')
df_500k_zip.head()

E:\Anaconda\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (10,13) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


id            date_submission  \
0  59f883d850b733300fdbec7a  2017-05-24T22:58:31+00:00   
1  59f883d850b733300fdbec7b  2017-05-24T22:58:31+00:00   
2  59f883d850b733300fdbec7c  2017-05-24T22:58:31+00:00   
3  59f883d850b733300fdbec7d  2017-05-24T22:58:31+00:00   
4  59f883d850b733300fdbec7e  2017-05-24T22:58:31+00:00   

             contact_email  confirmation_number  \
0       amos31wj@yahoo.com   201706031279605654   
1         dseanger@msn.com   201706030519826565   
2  loverlego7787@gmail.com     2017060327172116   
3       crankd84@gmail.com   201706030155900088   
4      lisadh728@yahoo.com    20170603026210221   

                                           text_data           index  \
0  Dear Chairman Pai,  I'm contacting you about r...  filings.2017.5   
1  Chairman Pai:  My comments re: net neutrality ...  filings.2017.5   
2  Mr Pai:  Hi, I'd like to comment on Internet f...  filings.2017.5   
3  Dear Chairman Pai,  I'm very worried about the...  filings.2017.5   
4  To the Federal Communications Commission:  Reg...  filings.2017.5   

                         filers  \
0    [{"name":"Amos Thompson"}]   
1  [{"name":"Douglas Seanger"}]   
2         [{"name":"Joe Cohn"}]   
3    [{"name":"Danelle Crank"}]   
4    [{"name":"Lisa Hockaday"}]   

                                       addressentity  \
0  {'city': 'W Jefferson', 'address_line_1': '133...   
1  {'city': 'Tucson', 'address_line_1': '1650 E R...   
2  {'city': 'Carol Stream', 'address_line_1': '25...   
3  {'city': 'Canon City', 'address_line_1': '2002...   
4  {'city': 'Raleigh', 'address_line_1': '1818 Wa...   

  internationaladdressentity           address_line_1 address_line_2  \
0         {"addresstext":""}           133 Blendon Rd            NaN   
1         {"addresstext":""}  1650 E River Rd Ste 104            NaN   
2         {"addresstext":""}   258 E Saint Charles Rd            NaN   
3         {"addresstext":""}           2002 E Main St            NaN   
4         {"addresstext":""}          1818 Watkins St            NaN   

           city state zip4 zip_code  
0   W Jefferson    OH  NaN    43162  
1        Tucson    AZ  NaN    56303  
2  Carol Stream    IL  NaN    60188  
3    Canon City    CO  NaN    81212  
4       Raleigh    NC  NaN    27604

In [7]:
# remove unwanted columns "address line 2" & "zip4"
df_500k_zip = df_500k_zip.drop(['address_line_2', 'zip4'], axis=1)

In [8]:
#save data
df_500k_zip.to_csv('FCC_comments_500K.csv', index=False)

In [28]:
# expand international address entity

df_500K_zip["internationaladdressentity"] = df_500K_zip["internationaladdressentity"].apply(lambda x : dict(eval(x)))
df2 = df_500K_zip["internationaladdressentity"].apply(pd.Series)

KeyError: 'internatioaladdressentity'

In [33]:
#concatonate dataframes
df_500k_zip = pd.concat([df_500K_zip, df2], axis=1)

In [39]:
df_500k_zip.tail()

id           date_submission  \
499995  59f88b9f50b733300fe38ed1  2017-05-25T21:34:18.557Z   
499996  59f88b9f50b733300fe38ed2  2017-05-25T21:34:18.558Z   
499997  59f88b9f50b733300fe38ed3  2017-05-25T21:34:18.714Z   
499998  59f88b9f50b733300fe38ed4  2017-05-25T21:34:18.725Z   
499999  59f88b9f50b733300fe38ed5  2017-05-25T21:34:18.746Z   

                    contact_email  confirmation_number  \
499995     BogdanTikhonov@mail.ru   201705251986403858   
499996     EmmaAlekseyeva@mail.ru    20170525154035820   
499997  MethodiusYevseyev@mail.ru    20170525032256819   
499998        EricBobrova@mail.ru     2017052570584704   
499999    StepanDegtyarev@mail.ru     2017052593883370   

                                                text_data           index  \
499995  We need net neutralityto continue. A free and ...  filings.2017.5   
499996  Save net neutrality! I am worried that the pro...  filings.2017.5   
499997  Leave the net neutrality alone. Internet speed...  filings.2017.5   
499998  Please Save Net Neutrality and Title ll rules....  filings.2017.5   
499999  Please save the internet from the corporations...  filings.2017.5   

                                 filers  \
499995     [{"name":"Bogdan Tikhonov"}]   
499996     [{"name":"Emma Alekseyeva"}]   
499997  [{"name":"Methodius Yevseyev"}]   
499998        [{"name":"Eric Bobrova"}]   
499999    [{"name":"Stepan Degtyarev"}]   

                                            addressentity  \
499995  {'city': '', 'address_line_1': '', 'state': ''...   
499996  {'city': '', 'address_line_1': '', 'state': ''...   
499997  {'city': '', 'address_line_1': '', 'state': ''...   
499998  {'city': '', 'address_line_1': '', 'state': ''...   
499999  {'city': '', 'address_line_1': '', 'state': ''...   

                               internationaladdressentity address_line_1 city  \
499995  {'addresstext': '3131 Haul Road
94015,Daly Cit...            NaN  NaN   
499996  {'addresstext': '3225 Joy Lane
91601,North Hol...            NaN  NaN   
499997  {'addresstext': '72 Mount Olive Road
30030,Dec...            NaN  NaN   
499998  {'addresstext': '888 Sherwood Circle
70501,Laf...            NaN  NaN   
499999  {'addresstext': '1379 Westfall Avenue
87102,Al...            NaN  NaN   

       state zip_code                         international_address_text  
499995   NaN      NaN  3131 Haul Road\n94015,Daly City, California, U...  
499996   NaN      NaN  3225 Joy Lane\n91601,North Hollywood, Californ...  
499997   NaN      NaN  72 Mount Olive Road\n30030,Decatur, Georgia, U...  
499998   NaN      NaN  888 Sherwood Circle\n70501,Lafayette, Louisian...  
499999   NaN      NaN  1379 Westfall Avenue\n87102,Albuquerque, New M...

In [38]:
#rename newly column from addresstext
df_500k_zip.rename(columns = {'addresstext':'international_address_text'}, inplace = True)

In [40]:
#save data
df_500k_zip.to_csv('FCC_comments_500K.csv', index=False)

In [3]:
#inspect data point
df_500k_zip.loc[499995]['international_address_text']

NameError: name 'df_500k_zip' is not defined

No surprise that
3131 Haul Road\n94015,Daly City, California
does not exist

In [2]:
df_500k_zip.loc[499496]['text_data']

NameError: name 'df_500k_zip' is not defined

In [76]:
# Multiple sumbissions at the exact same instant.. possibly automatically generated
print(df_500k_zip.date_submission.value_counts())

2017-05-24T23:18:04+00:00    86237
2017-05-24T23:09:20+00:00    81837
2017-05-24T23:14:50+00:00    61642
2017-05-24T23:11:57+00:00    38261
2017-05-24T23:19:47+00:00    30394
2017-05-25T10:01:58+00:00    13403
2017-05-24T22:58:31+00:00    12958
2017-05-25T10:01:36+00:00    12858
2017-05-24T23:21:10+00:00    12319
2017-05-25T10:01:15+00:00    12245
2017-05-25T10:03:36+00:00     9432
2017-05-25T10:02:58+00:00     9029
2017-05-25T10:00:52+00:00     8257
2017-05-25T10:02:15+00:00     7767
2017-05-25T10:04:06+00:00     7473
2017-05-25T10:02:36+00:00     7151
2017-05-25T10:00:32+00:00     6213
2017-05-25T09:59:45+00:00     5249
2017-05-25T10:00:10+00:00     5246
2017-05-25T10:03:50+00:00     4099
2017-05-25T10:03:18+00:00     3776
2017-05-25T21:15:41.088Z         3
2017-05-25T20:06:31.317Z         3
2017-05-25T20:36:58.727Z         3
2017-05-25T20:30:19.853Z         3
2017-05-25T14:23:11.888Z         2
2017-05-25T20:42:26.816Z         2
2017-05-25T20:36:04.326Z         2
2017-05-25T14:40:51.

Create small dataset to experiment with:

In [9]:
#create small df to experimen & train with
small_df = df_500K_zip[:501]

In [56]:
#inspect small_df
small_df.head(1)

id     date_submission       contact_email  \
0  59f883d850b733300fdbec7a 2017-05-24 22:58:31  amos31wj@yahoo.com   

   confirmation_number                                          text_data  \
0   201706031279605654  Dear Chairman Pai,  I'm contacting you about r...   

            index                      filers  \
0  filings.2017.5  [{"name":"Amos Thompson"}]   

                                       addressentity  \
0  {'city': 'W Jefferson', 'address_line_1': '133...   

  internationaladdressentity  address_line_1         city state zip_code  
0        {'addresstext': ''}  133 Blendon Rd  W Jefferson    OH    43162

In [55]:
small_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 501 entries, 0 to 500
Data columns (total 13 columns):
id                            501 non-null object
date_submission               501 non-null datetime64[ns]
contact_email                 501 non-null object
confirmation_number           501 non-null int64
text_data                     501 non-null object
index                         501 non-null object
filers                        501 non-null object
addressentity                 501 non-null object
internationaladdressentity    501 non-null object
address_line_1                501 non-null object
city                          501 non-null object
state                         501 non-null object
zip_code                      501 non-null object
dtypes: datetime64[ns](1), int64(1), object(11)
memory usage: 51.0+ KB


In [54]:
small_df['date_submission'] =  pd.to_datetime(small_df['date_submission'])

E:\Anaconda\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [ ]:
import plotly.plotly as py
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/2011_us_ag_exports.csv')

for col in df.columns:
    df[col] = df[col].astype(str)

scl = [[0.0, 'rgb(242,240,247)'],[0.2, 'rgb(218,218,235)'],[0.4, 'rgb(188,189,220)'],\
            [0.6, 'rgb(158,154,200)'],[0.8, 'rgb(117,107,177)'],[1.0, 'rgb(84,39,143)']]

df['text'] = df['state'] + '<br>' +\
    'Beef '+df['beef']+' Dairy '+df['dairy']+'<br>'+\
    'Fruits '+df['total fruits']+' Veggies ' + df['total veggies']+'<br>'+\
    'Wheat '+df['wheat']+' Corn '+df['corn']

data = [ dict(
        type='choropleth',
        colorscale = scl,
        autocolorscale = False,
        locations = df['code'],
        z = df['total exports'].astype(float),
        locationmode = 'USA-states',
        text = df['text'],
        marker = dict(
            line = dict (
                color = 'rgb(255,255,255)',
                width = 2
            )
        ),
        colorbar = dict(
            title = "Millions USD"
        )
    ) ]

layout = dict(
        title = '2011 US Agriculture Exports by State<br>(Hover for breakdown)',
        geo = dict(
            scope='usa',
            projection=dict( type='albers usa' ),
            showlakes = True,
            lakecolor = 'rgb(255, 255, 255)',
        ),
    )

fig = dict( data=data, layout=layout )

url = py.plot( fig, filename='d3-cloropleth-map' )

In [ ]:
## plot time series

In [3]:
df_3M = pd.read_csv('FCC_comments_3M.csv', encoding='utf8')

In [4]:
plt.hist(df_3M.date_submission)
plt.show()

ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.



Traceback (most recent call last):
  File "E:\Anaconda\lib\site-packages\IPython\core\interactiveshell.py", line 2862, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-93372b89c080>", line 1, in <module>
    plt.hist(df_3M.date_submission)
  File "E:\Anaconda\lib\site-packages\matplotlib\pyplot.py", line 3004, in hist
    stacked=stacked, normed=normed, data=data, **kwargs)
  File "E:\Anaconda\lib\site-packages\matplotlib\__init__.py", line 1710, in inner
    return func(ax, *args, **kwargs)
  File "E:\Anaconda\lib\site-packages\matplotlib\axes\_axes.py", line 6147, in hist
    x = self.convert_xunits(x)
  File "E:\Anaconda\lib\site-packages\matplotlib\artist.py", line 191, in convert_xunits
    return ax.xaxis.convert_units(x)
  File "E:\Anaconda\lib\site-packages\matplotlib\axis.py", line 1491, in convert_units
    ret = self.converter.convert(x, self.units, self)
  File "E:\Anaconda\lib\site-packages\matplotlib\category.py", line 51, in conv

KeyboardInterrupt: 